src/Tablebot/Plugins/Quote.hs

```diff
@@ -47,34 +71,143 @@ quoteComm ::
   WithError
     "Unknown quote functionality."
     ( Either
-        (Exactly "add", Quoted String, Exactly "-", RestOfInput String)
-        (Either (Exactly "show", Int) (Exactly "delete", Int))
+        ( Either
+            ( Either
+                (Exactly "add", Quoted String, Exactly "-", RestOfInput String)
+                (Exactly "edit", Int, Quoted String, Exactly "-", RestOfInput String)
+            )
+            ( Either
+                (Exactly "this")
+                (Exactly "user", RestOfInput String)
+            )
+        )
+        ( Either
+            ( Either
+                (Exactly "show", Int)
+                (Exactly "delete", Int)
+            )
+            ( Either
+                (Exactly "random")
+                ()
+            )
+        )
     ) ->
   Message ->
   DatabaseDiscord ()
```

Comment:
I promise we'll implement subcommands at some point - honestly it might be worth implementing subcommands before merging this in. (Thoughts appreciated.)
If it'd be useful, earliest I can do is Wednesday evening.